File: src/main/java/core/procedures/characters/IntegerToChar.java
package core.procedures.characters;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class IntegerToChar extends AFn {

  public IntegerToChar() {
    super(new FnArgsBuilder().min(1).max(1).mandatory(new Class[]{Long.class}).build());
  }

  @Override
  public String getName() {
    return "integer->char";
  }

  @Override
  public Character apply1(Object arg) {
    return (char)((Number)arg).longValue();
  }
}

File: src/main/java/core/procedures/characters/IntegerToChar.kt
package core.procedures.characters

import core.procedures.AFn
import core.procedures.Arity.Exactly

class IntegerToChar : AFn<Number?, Char>(name = "integer->char", arity = Exactly(1),
                                         mandatoryArgsTypes =  arrayOf(Long::class.javaObjectType)) {

    override operator fun invoke(arg: Number?) = arg!!.toChar()
}